judge/plan_entry_handler.go

```diff
@@ -0,0 +1,119 @@
+package judge
+
+import (
+	"sort"
+	"strconv"
+
+	"github.com/paketo-buildpacks/packit"
+)
+
+type Logger interface {
+	Subprocess(string, ...interface{})
+	Action(string, ...interface{})
+	Break()
+}
+
+type Emitter interface {
+	Candidates([]packit.BuildpackPlanEntry)
+}
+
+type PlanEntryHandler struct {
+	logger Logger
+}
+
+func NewPlanEntryHandler(logger Logger) PlanEntryHandler {
+	return PlanEntryHandler{
+		logger: logger,
+	}
+}
+
+func (p PlanEntryHandler) ResolveEntries(name string, entries []packit.BuildpackPlanEntry, priorities map[string]int) (packit.BuildpackPlanEntry, bool) {
+	var filteredEntries []packit.BuildpackPlanEntry
+	for _, e := range entries {
+		if e.Name == name {
+			filteredEntries = append(filteredEntries, e)
+		}
+	}
+
+	if len(filteredEntries) == 0 {
+		return packit.BuildpackPlanEntry{}, false
+	}
+
+	sort.Slice(filteredEntries, func(i, j int) bool {
+		leftSource := filteredEntries[i].Metadata["version-source"]
+		left, _ := leftSource.(string)
+
+		rightSource := filteredEntries[j].Metadata["version-source"]
+		right, _ := rightSource.(string)
+
+		return priorities[left] > priorities[right]
+	})
+
+	emitter, ok := p.logger.(Emitter)
+	if ok {
+		emitter.Candidates(filteredEntries)
+	} else {
+		p.candidates(filteredEntries)
+	}
+
+	return filteredEntries[0], true
```

Comment:
I wonder if you could simplify the API of this package by returning just a filtered and sorted list. It reduces some of the cognitive overhead of understanding the meaning attached to the `bool` return value, and negates the need to even deal with a `Logger` or `Emitter` type as those lines can be printed by the caller after they receive a filtered and sorted set of entries.

Reply:
I think that there are a couple of things that make me hesitant to do this, but maybe I am making too big a deal out of it. Firstly, I like the idea that if you are using this optional function we enforce logging in a format that we feel is appropriate. Secondly, picking an entry feels like it is better for the user in the long run than giving them a list back. If we just hand back a sorted filtered list I feel that there will be a lot of the same code following it which will be to the effect of:
```go
entries := planner.Resolve(name, context.Plan.Entries, priorities)
// bloat
logger.Candidates(entries)
entry := entries[0]
```
I do agree with the idea that the `bool` is not a great interface. Would it maybe make sense to have a typed error must like `os.ErrNotExists` so that it could still be checked?

Reply:
@ryanmoran if you want to check out my changes and see if that flows better to you.